soap/index.js

```diff
@@ -45,17 +45,62 @@ module.exports = yeoman.Base.extend({
 
   loadProject: actions.loadProject,
 
-  askForWsdlUrlOrPath: function() {
-    var prompts = [
-      {
-        name: 'url',
-        message: g.f('Enter the WSDL url or file path:'),
-        default: this.url,
-        validate: validateUrlOrFile,
-      },
-    ];
+  loadDataSources: actions.loadDataSources,
+
+  addNullDataSourceItem: actions.addNullDataSourceItem,
+
+  loadModels: actions.loadModels,
+
+  existingModels: function() {
+    var self = this;
+    self.existingModels = this.modelNames;
+  },
+
+  checkForDatasource: function() {
+    var self = this;
+    var soapDataSourceNames = [];
+    var soapDataSources = [];
+    for (var i in this.dataSources) {
```

Comment:
IIUC, `this.dataSources` is an array. Using `for...in` loop against an array is not practice. You should be able to use `this.dataSources.filter()` to get a subarray that matches `soap`.